lib/action_view/component/validatable.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module ActionView
+  module Component # :nodoc:
+    module Validatable
+      extend ActiveSupport::Concern
+
+      included do
+        if ActionView::Component.extend_validations
+          include ActiveModel::Validations
+        end
+
+        # method `validate!` actually performs validation and can be overrided
+        def validate_input!
+          return if self.class.validation_skiped?
+          validate!
+        end
+      end
```

Comment:
Why do we need the `validate_input!` wrapper vs just inlining in base as

```validate! unless skip_validation```

Reply:
```ruby
include ActionView::Component::Validatable
validate_input!
```

vs 

```ruby
include ActionView::Component::Validatable
validate! unless skip_validation
```

In first case you have to know about only one method `validate_input!` of related `Validatable` module in `Base`. While the second one requires you to know about `skip_validation` also. That increase code coupling